src/sourmash/sig/__main__.py

```diff
@@ -1283,6 +1291,98 @@ def fileinfo(args):
         print(json.dumps(info_d))
 
 
+def check(args):
+    """
+    check signature db(s) against a picklist.
+    """
+    from sourmash.picklist import PickStyle
+    set_quiet(args.quiet)
+    moltype = sourmash_args.calculate_moltype(args)
+    picklist = sourmash_args.load_picklist(args)
+    pattern_search = sourmash_args.load_include_exclude_db_patterns(args)
+    _extend_signatures_with_from_file(args)
+
+    if not picklist:
+        error("** No picklist provided?! Exiting.")
+        sys.exit(-1)
+
+    if picklist.pickstyle == PickStyle.EXCLUDE and args.output_missing:
+        error("** ERROR: Cannot use an 'exclude' picklist with '-o/--output-missing'")
+        sys.exit(-1)
+
+    # require manifests?
+    require_manifest = True
+    if args.no_require_manifest:
+        require_manifest = False
+        debug("sig check: manifest will not be required")
+    else:
+        debug("sig check: manifest required")
+
+    total_manifest_rows = []
+
+    # start loading!
+    total_rows_examined = 0
+    for filename in args.signatures:
+        idx = sourmash_args.load_file_as_index(filename,
+                                               yield_all_files=args.force)
+
+        idx = idx.select(ksize=args.ksize, moltype=moltype)
+
+        manifest = idx.manifest
+        if manifest is None:
+            if require_manifest:
+                error(f"ERROR on filename '{filename}'.")
+                error("sig check requires a manifest by default, but no manifest present.")
+                error("specify --no-require-manifest to dynamically generate one.")
+                sys.exit(-1)
+
+        manifest = sourmash_args.get_manifest(idx, rebuild=True)
```

Comment:
yep, nice catch. This was slightly tortured logic from some refactoring, and I've simplified it in a90a1b4a.

I, uhh, swear I leave these things in here so that reviewers have something to catch. Yeah, that's it! 😅

Reply:
Nice, that helps, thx!

I think I'm still a bit confused about line 1339 though... In my reading of the `get_manifest` code, `rebuild=True` forces rebuilding of the manifest. So in order to avoid rebuilding when you already have one, we need to set `rebuild=False` or only call this when manifest is None. What am I missing?

https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/sourmash_args.py#L779-L790
`if m is not None and not rebuild:` requires _both_ manifest AND rebuild=False, right?

Reply:
@ctb in case you don't get notified bc PR is closed :)

Reply:
you are correct, I think I messed that up. will revisit in a bit!

Reply:
fixed over in https://github.com/sourmash-bio/sourmash/pull/1915. thanks!!